invisible_cities/cities/components.py

```diff
@@ -152,6 +154,39 @@ def print_every_loop(target):
     return print_every_loop
 
 
+def collect():
+    """Return a future/sink pair for collecting streams into a list."""
+    def append(l,e):
+        l.append(e)
+        return l
+    return fl.reduce(append, initial=[])()
+
+
+def copy_mc_info(files_in : List[str], h5out: tb.File, event_numbers: List[int]):
+    """Copy to an output file the MC info of a list of selected events.
+
+    Parameters
+    ----------
+    files_in : List of strings
+        Name of the input files.
+    file_out : tables.File
+        The output h5 file.
+    event_numbers : List[int]
+        List of event numbers for which the MC info is copied to the output file.
+    """
+
+    writer = mcinfo_io.mc_info_writer(h5out)
+
+    for f in files_in:
+        with tb.open_file(f, "r") as h5in:
+            try:
+                event_numbers_in_file = h5in.root.MC.extents.cols.evt_number[:]
+                event_numbers_to_copy = set(event_numbers_in_file).intersection(event_numbers)
+                mcinfo_io.copy_mc_info(h5in, writer, list(event_numbers_to_copy))
+            except tb.exceptions.NoSuchNodeError:
```

Comment:
Maybe throwing a warning? Notice that we want to be able to run the cities on empty files

Reply:
Ah, if that is the reason I guess it makes sense. But maybe we can introduce a function to detect empty files and use that. With this implementation we cannot distinguish between that and a file that has something wrong.

Reply:
> Ah, if that is the reason I guess it makes sense. But maybe we can introduce a function to detect empty files and use that. With this implementation we cannot distinguish between that and a file that has something wrong.

That is the reason, yes.

Reply:
@mmkekic., do you know whether an empty file contains the `runInfo` table with zero entries? This could be the key to detect empty files vs "corrupt" files.

Reply:
They do! So maybe thats the first thing we can check per file_in, if its empty do nothing, if its not and the reading doesnt work raise an error

Reply:
The test `cities_test.test_city_empty_input_file` uses an empty input file that contains no tables.

Reply:
I can't confirm, but I would say that this is not realistic. It depends on the decoder, which is not in this repo, but I think these tables are always there, even if empty. In any case, a function that checks whether a file is empty or not could check both whether they exist and, if so, check whether they contain any entries.

Reply:
Sure, but shouldn't that be a task for a different PR?

Reply:
I would prefer that, but that would delay this one, right? I am open to introducing it here, if necessary. @mmkekic ?

Reply:
I just want to accept this PR so even a warning would be fine for me :p
But if you dont want to do sloppy job add a function that checks if the file is empty in this PR

Reply:
> I just want to accept this PR so even a warning would be fine for me :p
I'd honestly follow this approach. Print a warning for now and leave as a to-do the "empty-file" checks, which could even require modifying the existing tests. I would even volunteer to do that…

Reply:
Fine, go ahead.